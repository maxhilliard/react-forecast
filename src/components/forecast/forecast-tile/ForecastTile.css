.forecastTile {
    display: grid;
    grid-template-columns: 33% 33% 33%;
    grid-template-rows: 50% 50%;
    grid-template-areas:
        "forecastDay forecastIcon forecastTemps"
        "forecastDay forecastIcon forecastHumidity";
}

.forecastDay {
    grid-area: forecastDay;
}

.forecastIcon {
    grid-area: forecastIcon;
}

.forecastTemps {
    grid-area: forecastTemps;
}

.forecastHumidity {
    grid-area: forecastHumidity;
}