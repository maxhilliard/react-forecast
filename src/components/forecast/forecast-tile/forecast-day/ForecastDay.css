.forecastDay {
    composes: forecastDay from '../ForecastTile.css';
    font-size: 30px;
    margin: 10px;
}