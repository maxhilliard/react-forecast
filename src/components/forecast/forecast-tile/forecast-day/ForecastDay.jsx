import React from 'react';
import PropTypes from 'prop-types';

import styles from './ForecastDay.css';

const ForecastDay = ({ day }) => (
    <p className={styles.forecastDay}>
        {day}
    </p>
);

ForecastDay.propTypes = {
    day: PropTypes.string.isRequired,
};

export default ForecastDay;
