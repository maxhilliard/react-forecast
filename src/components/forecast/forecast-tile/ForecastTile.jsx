import React from 'react';
import PropTypes from 'prop-types';
import { format } from 'date-fns';

import ForecastDay from './forecast-day/ForecastDay';
import ForecastIcon from './forecast-icon/ForecastIcon';
import ForecastTemps from './forecast-temps/ForecastTemps';
import ForecastHumidity from './forecast-humidity/ForecastHumidity';

import styles from './ForecastTile.css';

// TODO - move this parsing into util
const parseDayFromUnixTimestamp = timestamp => format(timestamp * 1000, 'ddd');

const ForecastTile = ({
    dt,
    humidity,
    temp,
    weather,
}) => {
    const day = parseDayFromUnixTimestamp(dt);
    const { icon, description } = weather[0];

    return (
        <div className={styles.forecastTile}>
            <ForecastDay day={day} />
            <ForecastIcon icon={icon} description={description} />
            <ForecastTemps temp={temp} />
            <ForecastHumidity humidity={humidity} />
        </div>
    );
};

ForecastTile.propTypes = {
    dt: PropTypes.number.isRequired,
    humidity: PropTypes.number.isRequired,
    temp: PropTypes.objectOf(PropTypes.number).isRequired,
    weather: PropTypes.array.isRequired, // eslint-disable-line
};

export default ForecastTile;
