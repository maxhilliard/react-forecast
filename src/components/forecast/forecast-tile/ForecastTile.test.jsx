import React from 'react';
import { shallow } from 'enzyme';

import ForecastTile from './ForecastTile';
import ForecastDay from './forecast-day/ForecastDay';

describe('ForecastTile', () => {
    const mockForecastTileProps = {
        dt: 1540078790,
        humidity: 70,
        temp: { min: 12, max: 23 },
        weather: ['mock-weather'],
    };

    const wrapper = shallow(<ForecastTile {...mockForecastTileProps} />);

    describe('snapshot', () => {
        it('should match snapshot', () => {
            expect(wrapper).toMatchSnapshot();
        });
    });

    describe('props', () => {
        it('should parse dt from a unix timestamp to a day value', () => {
            const parsedDay = wrapper
                .find(ForecastDay)
                .props()
                .day;

            const expectedDay = 'Sun';

            expect(parsedDay).toEqual(expectedDay);
        });
    });
});
