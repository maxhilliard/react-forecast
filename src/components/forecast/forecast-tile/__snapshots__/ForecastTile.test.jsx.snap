// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ForecastTile snapshot should match snapshot 1`] = `
<div
  className="forecastTile"
>
  <ForecastDay
    day="Sun"
  />
  <ForecastIcon />
  <ForecastTemps
    temp={
      Object {
        "max": 23,
        "min": 12,
      }
    }
  />
  <ForecastHumidity
    humidity={70}
  />
</div>
`;
